solr/core/src/java/org/apache/solr/schema/IndexSchema.java

```diff
@@ -116,6 +116,18 @@ public class IndexSchema {
   private static final String SOURCE_DYNAMIC_BASE = "sourceDynamicBase";
   private static final String SOURCE_EXPLICIT_FIELDS = "sourceExplicitFields";
 
+  /**
+   * Implicit default for `uninvertible` FieldType property, if not specified by `fieldType` or
+   * `field`. This implicit default may be overridden by a node-level default specified in
+   * `solr.xml`. The implicit defaults is <code>true</code> for historical reasons, but users are
+   * strongly encouraged to set this to <code>false</code> for stability and use <code>
+   * docValues="true"</code> or <code>uninvertible="true"</code> as needed.
+   */
+  public static final UninvertingReader.Support IMPLICIT_DEFAULT_UNINVERTIBLE =
+      UninvertingReader.Support.DEFAULT_TRUE;
+
+  private static UninvertingReader.Support uninvertibleSupport = IMPLICIT_DEFAULT_UNINVERTIBLE;
```

Comment:
static non-final variables concern me, as they should anyone.  It has to be handled carefully in tests to ensure it's reset.  If we can avoid this, we should.  Did you investigate passing this into the IndexSchema constructor, ultimately held in the NodeConfig?

Reply:
I'll look into this (I think this ties into a number of your other comments as well). I was following the pattern of `maxBooleanClauses` here, but I agree if this can be done another way, that would be preferable.